include/envoy/http/async_client.h

```diff
@@ -140,8 +140,14 @@ class AsyncClient {
    *         onFailure() has already been called inline. The client owns the request and the
    *         handle should just be used to cancel.
    */
-  virtual Request* send(MessagePtr&& request, Callbacks& callbacks,
-                        const absl::optional<std::chrono::milliseconds>& timeout) PURE;
+  struct SendArgs {
```

Comment:
As these are passing down to AsyncRequest/AsyncStream, should named RequestOptions/StreamOptions, and perhaps inheritance?

Reply:
+1

Reply:
Done